src/util/mp_arith.h

```diff
@@ -62,4 +62,18 @@ unsigned integer2unsigned(const mp_integer &);
 
 const mp_integer mp_zero=string2integer("0");
 
+/// Minimum of an mp_integer and another integer value
+template <typename T>
+mp_integer min(const mp_integer &m, const T& n)
+{
+  return m <= n ? m : mp_integer(n);
+}
+
+/// Maximum of an mp_integer and another integer value
+template <typename T>
+mp_integer max(const mp_integer &m, const T& n)
+{
+  return m >= n ? m : mp_integer(n);
+}
```

Comment:
Let's just not do this. We don't have to reinvent every library function with just a tiny tweak to its API.